src/views/index-js.mustache

```diff
@@ -5169,47 +5169,25 @@ function getSize (size) {
-    }
-    const femaleSuffix = female ? '_female' : '';
-    if (pokemonForm) {
-        let pokemonIdString = `${pokeId}_${pokemonForm}${femaleSuffix}${costumeSuffix}${shinySuffix}`;
-        if (availableForms.has(pokemonIdString)) return pokemonIdString;
-        pokemonIdString = `${pokeId}_${pokemonForm}${costumeSuffix}${shinySuffix}`;
-        if (availableForms.has(pokemonIdString)) return pokemonIdString;
-        pokemonIdString = `${pokeId}_${pokemonForm}${femaleSuffix}${shinySuffix}`;
-        if (availableForms.has(pokemonIdString)) return pokemonIdString;
-        pokemonIdString = `${pokeId}_${pokemonForm}${shinySuffix}`;
-        if (availableForms.has(pokemonIdString)) return pokemonIdString;
-    }
-    const defaultForm = (masterfile.pokemon[pokemonId] || {}).default_form_id;
-    if (defaultForm) {
-        let pokemonIdString = `${pokeId}_${defaultForm}${femaleSuffix}${costumeSuffix}${shinySuffix}`;
-        if (availableForms.has(pokemonIdString)) return pokemonIdString;
-        pokemonIdString = `${pokeId}_${defaultForm}${costumeSuffix}${shinySuffix}`;
-        if (availableForms.has(pokemonIdString)) return pokemonIdString;
-        pokemonIdString = `${pokeId}_${defaultForm}${femaleSuffix}${shinySuffix}`;
-        if (availableForms.has(pokemonIdString)) return pokemonIdString;
-        pokemonIdString = `${pokeId}_${defaultForm}${shinySuffix}`;
-        if (availableForms.has(pokemonIdString)) return pokemonIdString;
-    }
-    let pokemonIdString = `${pokeId}_00${femaleSuffix}${costumeSuffix}${shinySuffix}`;
-    if (availableForms.has(pokemonIdString)) return pokemonIdString;
-    pokemonIdString = `${pokeId}_00${costumeSuffix}${shinySuffix}`;
-    if (availableForms.has(pokemonIdString)) return pokemonIdString;
-    pokemonIdString = `${pokeId}_00${femaleSuffix}${shinySuffix}`;
-    if (availableForms.has(pokemonIdString)) return pokemonIdString;
-    pokemonIdString = `${pokeId}_00${shinySuffix}`;
-    if (availableForms.has(pokemonIdString)) return pokemonIdString;
-    return '000';   // substitute
+function getPokemonIcon(pokemonId, form = 0, evolution = 0, gender = 0, costume = 0, shiny = false) {
+    const evolutionSuffixes = evolution ? ['-e' + evolution, ''] : [''];
+    const formSuffixes = form ? ['-f' + form, ''] : [''];
+    const costumeSuffixes = costume ? ['-c' + costume, ''] : [''];
+    const genderSuffixes = gender ? ['-g' + gender, ''] : [''];
+    const shinySuffixes = shiny ? ['-shiny', ''] : [''];
+    for (const evolutionSuffix of evolutionSuffixes) {
+    for (const formSuffix of formSuffixes) {
+    for (const costumeSuffix of costumeSuffixes) {
+    for (const genderSuffix of genderSuffixes) {
+    for (const shinySuffix of shinySuffixes) {
+        const result = `${pokemonId}${evolutionSuffix}${formSuffix}${costumeSuffix}${genderSuffix}${shinySuffix}`;
+        if (availableForms.has(result)) return result;
+    }
+    }
+    }
+    }
+    }
+    return '0'; // substitute
```

Comment:
😂  Oh god the formatting

Reply:
I think it looks nicer this way. :wink:

Reply:
> I think it looks nicer this way. 😉

I actually do agree tbh, C# has something similar i.e:
```cs
using (var evolutionSuffix in evolutionSuffixes) {
using (var formSuffix in formSuffixes) {
using (var costumeSuffix in costumeSuffixes) {
using (var genderSuffix in genderSuffixes) {
using (var shinySuffix in shinySuffixes) {
    var result = $`{pokemonId}{evolutionSuffix}{formSuffix}{costumeSuffix}{genderSuffix}{shinySuffix}`;
    if (availableForms.Contains(result)) return result;
```

Reply:
Please no 😜
Use indentation

Reply:
I don't understand. They look identical.

Reply:
Consistency though :P

Reply:
@versx already added an eslint file, so please be consistent with the formatting rules

Reply:
Maybe we should disable eslint on this function? I think an exception could/should be made in this case.